### ISSUE TITLE
ldc 1.9.0 fails on dmd-testsuite*/runnable/test17559.d

### ISSUE THREAD

**Issue by ggardet** [ldc-developers/ldc#2730]
On openSUSE Tumbleweed x86_64, LDC 1.9.0 fails on dmd-testsuite/runnable/test17559.d with the following trace:
```
 ... runnable/test17559.d           -g -L-export-dynamic -O ()
Test failed.  The logged output:
/home/abuild/rpmbuild/BUILD/ldc-1.9.0-src/build/bin/ldmd2 -conf= -m64 -Irunnable -g -L-export-dynamic -O  -od../../../build/dmd-testsuite/runnable -of../../../build/dmd-testsuite/runnable/test17559_0  runnable/test17559.d 
../../../build/dmd-testsuite/runnable/test17559_0
core.exception.AssertError@runnable/test17559.d(55): Assertion failure
----------------
??:? _d_assert [0x7f84ec28b0a5]
??:? void test17559.fun(int, int) [0x55c0dca92317]
??:? _Dmain [0x55c0dca921e8]
object.Exception@runnable/test17559.d(48): fun
----------------
??:? void test17559.fun(int, int) [0x55c0dca922b5]
??:? _Dmain [0x55c0dca921e8]


==============================
Test failed: expected rc == 0, exited with rc == 1

make[3]: *** [Makefile:423: ../../../build/dmd-testsuite/runnable/test17559.d.out] Error 1
```
and on dmd-testsuite-debug/runnable/test17559.d:
```
 ... runnable/test17559.d           -g -L-export-dynamic -g -link-defaultlib-debug ()
Test failed.  The logged output:
/home/abuild/rpmbuild/BUILD/ldc-1.9.0-src/build/bin/ldmd2 -conf= -m64 -Irunnable -g -L-export-dynamic -g -link-defaultlib-debug  -od../../../build/dmd-testsuite-debug/runnable -of../../../build/dmd-testsuite-debug/runnable/test17559_0  runnable/test17559.d 
../../../build/dmd-testsuite-debug/runnable/test17559_0
core.exception.AssertError@runnable/test17559.d(55): Assertion failure
----------------
??:? onAssertError [0x7f77ad8749e9]
??:? _d_assert [0x7f77ad8750b2]
??:? void test17559.fun(int, int) [0x55cdd674f3a8]
??:? _Dmain [0x55cdd674f20f]
object.Exception@runnable/test17559.d(48): fun
----------------
??:? void test17559.fun(int, int) [0x55cdd674f2fb]
??:? _Dmain [0x55cdd674f20f]


==============================
Test failed: expected rc == 0, exited with rc == 1

make[3]: *** [Makefile:423: ../../../build/dmd-testsuite-debug/runnable/test17559.d.out] Error 1
```

**Comment by ggardet** [comment]
Ok, thanks. 
I will test with v1.10 then.

**Comment by kinke** [comment]
@ggardet: Any news on this?

**Comment by ggardet** [comment]
@kinke Yes, I confirm this is fixed in v1.10. 
Thanks!
